lib/Url.js

```diff
@@ -11,6 +11,12 @@ const URL_REGEX = `(${URL_WEBSITE_REGEX}${URL_PATH_REGEX}(?:${URL_PARAM_REGEX}|$
 
 const URL_REGEX_WITH_REQUIRED_PROTOCOL = URL_REGEX.replace(`${URL_PROTOCOL_REGEX}?`, URL_PROTOCOL_REGEX);
 
+const LOOSE_URL_WEBSITE_REGEX = `${URL_PROTOCOL_REGEX}([-\\w]+(\\.[-\\w]+)*)(?:\\:${ALLOWED_PORTS}|\\b|(?=_))`;
```

Comment:
Can you explain this chunk of the regex please? How did you come up with it and what is it supposed to do?
```javascript
([-\\w]+(\\.[-\\w]+)*)
```

My understanding is that it should match the domain name, however it isn't explained that way on [regex101](https://regex101.com/r/1EKXMm/1). Maybe I'm missing something...

Reply:
Ah got it thanks!

> You need to replace the `\\` with `\` when testing it on `regex101`

Can you explain why?